### PR TITLE
chore: upgrade rust toolchain to 1.67.0-nightly

### PR DESCRIPTION
**Pull request by hanabi1224**
**Summary of changes**
Changes introduced in this pull request:
- Upgrade rust toolchain from `1.66.0-nightly` to `1.67.0-nightly`
- Fix `cargo clippy` errors

[1.65.0 changelog](https://releases.rs/docs/released/1.65.0/)

**Reference issue to close (if applicable)**

Closes 

**Other information and links**

### CODE REVIEW

File: .cargo/config.toml [hunk 1]
```diff
@@ -23,7 +23,7 @@ rustflags = ["-Clink-arg=-fuse-ld=lld", "-Ctarget-feature=+avx2,+fma", "-Zshare-
 # brew install michaeleisel/zld/zld
 # "-Clink-arg=-fuse-ld=zld"
 # For details checkout https://jondot.medium.com/8-steps-for-troubleshooting-your-rust-build-times-2ffc965fd13e
-rustflags = ["-Ctarget-feature=+avx2,+fma", "-Zshare-generics=y"]
```
Comment (hanabi1224) [inline .cargo/config.toml:26]: This causes build failure after upgrading libp2p

**Review by tyshko5 — APPROVED** [review]

**Review by LesnyRumcajs — APPROVED** [review]
LGTM

File: forest/cli/src/cli/net_cmd.rs [hunk 2]
```diff
@@ -78,7 +78,7 @@ impl NetCommands {
                 let addr: Multiaddr = address
                     .parse()
                     .map_err(|e| {
-                        cli_error_and_die(&format!("Error parsing multiaddr. Error was: {}", e), 1);
+                        cli_error_and_die(format!("Error parsing multiaddr. Error was: {}", e), 1);
```
[the remark this answers is not in the document]
Comment (hanabi1224) [reply forest/cli/src/cli/net_cmd.rs]: fixed

File: forest/cli/src/cli/wallet_cmd.rs [hunk 3]
```diff
@@ -226,7 +226,7 @@ impl WalletCommands {
                 let address_result = Address::from_str(address);
 
                 if address_result.is_err() {
-                    cli_error_and_die(&format!("{} is not a valid address", address), 1);
+                    cli_error_and_die(format!("{} is not a valid address", address), 1);
```
[the remark this answers is not in the document]
Comment (hanabi1224) [reply forest/cli/src/cli/wallet_cmd.rs]: fixed

File: forest/cli/src/cli/wallet_cmd.rs [hunk 4]
```diff
@@ -154,7 +154,7 @@ impl WalletCommands {
                     serde_json::from_str(key_str);
 
                 if key_result.is_err() {
-                    cli_error_and_die(&format!("{} is not a valid key to import", key), 1);
+                    cli_error_and_die(format!("{} is not a valid key to import", key), 1);
```
[the remark this answers is not in the document]
Comment (hanabi1224) [reply forest/cli/src/cli/wallet_cmd.rs]: fixed

File: forest/cli/src/cli/wallet_cmd.rs [hunk 5]
```diff
@@ -130,7 +130,7 @@ impl WalletCommands {
                 let key = match path {
                     Some(path) => match read_file_to_string(&PathBuf::from(path)) {
                         Ok(key) => key,
-                        _ => cli_error_and_die(&format!("{} is not a valid path", path), 1),
+                        _ => cli_error_and_die(format!("{} is not a valid path", path), 1),
```
[the remark this answers is not in the document]
Comment (hanabi1224) [reply forest/cli/src/cli/wallet_cmd.rs]: fixed

File: forest/cli/src/cli/state_cmd.rs [hunk 6]
```diff
@@ -89,7 +89,7 @@ impl StateCommands {
                         }
                     }
                     None => cli_error_and_die(
-                        &format!("cannot find miner at address {}", miner_address),
+                        format!("cannot find miner at address {}", miner_address),
```
[the remark this answers is not in the document]
Comment (hanabi1224) [reply forest/cli/src/cli/state_cmd.rs]: fixed

**Pull request merged**